sdk/core/Azure.Core/src/AzureKeyCredential.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ComponentModel;
+using System.Threading;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// API key credential used to authenticate to an Azure Service.
+    /// It provides the ability to update the API key without creating a new client.
+    /// </summary>
+    public class AzureKeyCredential
+    {
+        private string _apiKey;
+
+        /// <summary>
+        /// API key used to authenticate to an Azure service.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public string ApiKey
```

Comment:
Is this because the policy is shared source?

Reply:
Maybe we should just ship it as public API.

Reply:
We've been kind of lazy in how we hide these properties on other credential types.  There's no meaningful security benefit, but we're trying to guide users to not rely on reading keys from our types and instead be thoughtful about how they manage secure configuration details in their application.  I think the `EditorBrowsable` approach here is a clever way of walking that fine line by keeping our scenarios possible while still hiding it from the type of dev who might be tempted to misuse it.

Reply:
We need to decide if we want it public or not, if it is we shouldn't have a method that's basically a setter for the same property.

Do we have a scenario where this property is worth hiding? Because people who tend to missuse things usually find ways to do it and I doubt EditorBrowsable would stop them.

Reply:
>we shouldn't have a method that's basically a setter for the same property.

You mean the `Update` method? We want to be able to rotate the key without restarting clients, which is why we have the method available.

Reply:
Right, but having a public setter on the property would accomplish the same thing.

Reply:
A property setter would work here, but not for a lot of other credential types.  An `Update` method works better if you need to do any computation on the values, if you have multiple values that need to be swapped atomically, etc.  This is more about following @schaabs's Credential pattern even though the `Update` method isn't as necessary with this simplest of credentials.

Reply:
True. The [guidelines](https://azure.github.io/azure-sdk/general_design.html#authentication) say:
> DO define an update method which accepts all mutable credential data, and updates the credential in an atomic, thread safe manner.

So following just in case later we need to make modifications to the key